### PR TITLE
meson: scope more git invocations with current_source_dir()

### PR DESCRIPTION
**Pull request by davide125**
This is a followup to #6635 to replace more uses of `meson.source_root()` with `meson.current_source_dir()`, which will return the correct path when building systemd as a meson subproject.

### CODE REVIEW

**Comment by davide125** [comment]
Force pushed a version with fixes to the relative paths under `man/` and `test/fuzz/`, let's see if this is enough to make the CI happy.

File: man/meson.build [hunk 1]
```diff
@@ -20,7 +20,7 @@ xsltproc_flags = [
         '--stringparam', 'man.copyright.section.enabled', '0',
         '--stringparam', 'systemd.version', '@0@'.format(meson.project_version()),
         '--path',
-        '@0@:@1@'.format(meson.current_build_dir(), meson.current_source_dir())]
+        '@0@:@1@/..'.format(meson.current_build_dir(), meson.current_source_dir())]
```
Comment (keszybz) [inline man/meson.build]: This looks iffy.

File: test/fuzz/meson.build [hunk 2]
```diff
@@ -16,11 +16,12 @@ sanitizers = [['address', sanitize_address]]
 if git.found()
         out = run_command(
                 git,
-                '--git-dir=@0@/.git'.format(meson.source_root()),
+                '--git-dir=@0@/../../.git'.format(meson.current_source_dir()),
```
Comment (keszybz) [inline test/fuzz/meson.build]: This seems quite fragile. Maybe add a `project_source_root` variable in the root `meson.build` file and use it here? We tend to move files in the directory structure quite often, and relative paths with '..' is something that might become wrong with nobody noticing for a while.

**Comment by keszybz** [comment]
CI still fails. Also needs a rebase.

**Comment by davide125** [comment]
Rebased and updated using `project_source_root` to @keszybz suggestion. Still trying to figure out what's wrong with the CI, that failure doesn't repro locally for me (neither on Debian nor on CentOS).

**Comment by davide125** [comment]
Ok after fixing up a few more callsites the CI looks happier

**Comment by davide125** [comment]
bionic-i386 failure looks unrelated, everything else is passing now

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**